### PR TITLE
Added ESDT support on proxy

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added ESDT support on proxy

### CODE REVIEW

File: blockchain/proxy_test.go [hunk 1]
```diff
@@ -644,3 +663,227 @@ func TestProxy_GetValidatorsInfoByEpoch(t *testing.T) {
...
+		assert.Equal(t, responseTokenData, tokenData)
+		assert.False(t, responseTokenData == tokenData) // pointer testing
+	})
+}
+
+func TestElrondProxy_GetNFTTokenData(t *testing.T) {
+	t.Parallel()
+
+	token := "TKN-001122"
+	nonce := uint64(37)
```
Comment (bogdan-rosianu) [inline blockchain/proxy_test.go:779]: :+1:

File: blockchain/proxy.go [hunk 2]
```diff
@@ -559,6 +559,58 @@ func (ep *proxy) GetValidatorsInfoByEpoch(ctx context.Context, epoch uint32) ([]
 	return response.Data.ValidatorsInfo, nil
 }
 
+// GetESDTTokenData returns the address' fungible token data
+func (ep *proxy) GetESDTTokenData(ctx context.Context, address erdgoCore.AddressHandler, tokenIdentifier string) (*data.ESDTFungibleTokenData, error) {
```
Comment (bogdan-rosianu) [inline blockchain/proxy.go:563]: can be only a TODO for now, but we also support block coordinates in various endpoints (for example: "?blockNonce=373737"). We might add support for this as well in the sdk(s)

**Review by matei-p — DISMISSED** [review]

File: blockchain/proxy.go (hunk 2 above)
Comment (iulianpascalau) [reply blockchain/proxy.go:563]: done & added TODO for the other routes. Also, added TODO to move the function in the core repo and remove it from SDK & proxy.

**Review by matei-p — APPROVED** [review]

File: blockchain/proxy.go [hunk 3]
```diff
@@ -559,6 +560,73 @@ func (ep *proxy) GetValidatorsInfoByEpoch(ctx context.Context, epoch uint32) ([]
...
+) (*data.ESDTFungibleTokenData, error) {
+	if check.IfNil(address) {
+		return nil, ErrNilAddress
+	}
+	if !address.IsValid() {
+		return nil, ErrInvalidAddress
+	}
+
+	endpoint := ep.endpointProvider.GetESDTTokenData(address.AddressAsBech32String(), tokenIdentifier)
+	endpoint = erdgoCore.BuildUrlWithAccountQueryOptions(endpoint, queryOptions)
```
Comment (camilbancioiu) [inline blockchain/proxy.go:578]: Consider replacing `erdgoCore` with `mxCore`.

Comment (camilbancioiu) [reply blockchain/proxy.go:578]: And `erdgoHttp` as well.

**Review by camilbancioiu — APPROVED** [review]

File: blockchain/proxy.go (hunk 3 above)
Comment (bogdan-rosianu) [reply blockchain/proxy.go:578]: :+1:

Comment (iulianpascalau) [reply blockchain/proxy.go:578]: good point. Will search through the entire project.

**Review by AlexandruMarian2023 — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Review by matei-p — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**